lib/ansible/modules/cloud/amazon/elasticache_replication_group.py

```diff
@@ -0,0 +1,586 @@
+                         " for more details")
+        module.fail_json_aws(e, msg=msg)
+
+    if module.params.get('wait'):
+        waiter = connection.get_waiter('replication_group_available')
+        waiter.wait(ReplicationGroupId=params['ReplicationGroupId'])
+
+    rep_group = check_for_rep_group(module, connection)
+    module.exit_json(changed=True, **camel_dict_to_snake_dict(rep_group))
+
+
+def destroy_rep_group(module, connection):
+
+    params = dict()
+
+    params['ReplicationGroupId'] = module.params.get('name')
+    params['RetainPrimaryCluster'] = module.params.get('retain_primary_cluster')
+    params['FinalSnapshotIdentifier'] = module.params.get('final_snapshot_id')
+
+    # Remove any items with a value of None
+    for k, v in list(params.items()):
+        if v is None:
+            del params[k]
+
+    try:
+        connection.delete_replication_group(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't delete replication group")
+
+    if module.params.get('wait'):
+        waiter = connection.get_waiter('replication_group_deleted')
+        waiter.wait(ReplicationGroupId=params['ReplicationGroupId'])
+        module.exit_json(changed=True)
+    else:
+        module.exit_json(changed=True)
+
+
+def reboot_rep_group(module, connection, rep_group):
+    for node_group in rep_group['NodeGroups'][0]['NodeGroupMembers']:
+        try:
+            connection.reboot_cache_cluster(CacheClusterId=node_group['CacheClusterId'],
+                                            CacheNodeIdsToReboot=[node_group['CacheNodeId']])
+
+            changed = True
+            if module.params.get('wait'):
+                waiter = connection.get_waiter('cache_cluster_available')
+                waiter.wait(CacheClusterId=node_group['CacheClusterId'])
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, "Couldn't reboot cache cluster %s" % node_group['CacheClusterId'])
+
+    result = check_for_rep_group(module, connection)
+    module.exit_json(changed=changed, **camel_dict_to_snake_dict(result))
+
+
+def modify_rep_group(module, connection, rep_group):
+
+    changed = False
+
+    params = dict()
+    warnings = list()
+
+    # The output keys of describe_replication_groups are different
+    # to the input keys of modify_replication_group
+    rep_group['ReplicationGroupDescription'] = rep_group['Description']
+
+    rep_group_params = {
+        'description': 'ReplicationGroupDescription',
+        'primary_cluster_id': 'PrimaryClusterId',
+        'node_type': 'CacheNodeType',
+    }
+    cache_cluster_params = {
+        'cache_engine_version': 'EngineVersion',
+        'cache_parameter_group': 'CacheParameterGroupName',
+        'cache_subnet_group_name': 'CacheSubnetGroupName',
```

Comment:
Ok, I found another issue and this time I do have documentation to back me up. 😄 

 `CacheSubnetGroupName` does not exist as a modifiable parameter: http://boto3.readthedocs.io/en/latest/reference/services/elasticache.html#ElastiCache.Client.modify_replication_group